packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+      }
+      break;
+    }
+
+    if (foundHtmlTag.type === 'open' || foundHtmlTag.type === 'self-closing') {
+      const nextTag = parseOpenTag(foundHtmlTag.content, foundHtmlTag.startIdx);
+      // Add the plain text between the last tag and this one found
+      plainTextRepresentation += text.substring(parseIndex, foundHtmlTag.startIdx);
+      nextTag.plainTextBeginIndex = plainTextRepresentation.length;
+
+      if (foundHtmlTag.type === 'open') {
+        tagParseStack.push(nextTag);
+      } else {
+        nextTag.content = '';
+        nextTag.plainTextBeginIndex = plainTextRepresentation.length;
+        nextTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(nextTag, tagParseStack, tags);
+      }
+    }
+
+    if (foundHtmlTag.type === 'close') {
+      const currentOpenTag = tagParseStack.pop();
+      const closeTagType = foundHtmlTag.content.substring(2, foundHtmlTag.content.length - 1).toLowerCase();
+
+      if (currentOpenTag && currentOpenTag.tagType === closeTagType) {
+        // Tag startIdx is absolute to the text. This is updated later to be relative to the parent tag
+        currentOpenTag.content = text.substring(
+          currentOpenTag.openTagIdx + currentOpenTag.openTagBody.length,
+          foundHtmlTag.startIdx
+        );
+
+        // Insert the plain text pieces for the sub tags
+        if (currentOpenTag.tagType === 'msft-mention') {
+          plainTextRepresentation =
+            plainTextRepresentation.slice(0, currentOpenTag.plainTextBeginIndex) +
+            trigger +
+            plainTextRepresentation.slice(currentOpenTag.plainTextBeginIndex);
+        }
+
+        if (!currentOpenTag.subTags) {
+          plainTextRepresentation += currentOpenTag.content;
+        } else if (currentOpenTag.subTags.length > 0) {
+          // Add text after the last tag
+          const lastSubTag = currentOpenTag.subTags[currentOpenTag.subTags.length - 1];
+          const startOfRemainingText =
+            (lastSubTag.closeTagIdx ?? lastSubTag.openTagIdx) + lastSubTag.tagType.length + 3;
+          const trailingText = currentOpenTag.content.substring(startOfRemainingText);
+          plainTextRepresentation += trailingText;
+        }
+
+        currentOpenTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(currentOpenTag, tagParseStack, tags);
+      } else {
+        console.error(
+          'Unexpected close tag found. Got "' +
+            closeTagType +
+            '" but expected "' +
+            tagParseStack[tagParseStack.length - 1]?.tagType +
+            '"'
+        );
+      }
+    }
+
+    // Update parsing index; move past the end of the close tag
+    parseIndex = foundHtmlTag.startIdx + foundHtmlTag.content.length;
+  } // While parseIndex < text.length loop
+
+  return [tags, plainTextRepresentation];
+};
+
+/* @conditional-compile-remove(mention) */
+const parseOpenTag = (tag: string, startIdx: number): TagData => {
+  const tagType = tag
```

Comment:
Strings like:

< div /> or 
< hello > should be considered as just plain string instead of tags, do we handle cases like this?

Reply:
It's like space before after the open tag

Reply:
Negative testing also needed for sure; part of the test tasks.